### PR TITLE
Add event options in SDK

### PR DESCRIPTION
**Pull request by bencroker**
Adds event options to the `option` argument of all `ServerSentEventGenerator` helper methods and adds line breaks to make them easier to read.

### CODE REVIEW

File: design/SDK.md [hunk 1]
```diff
@@ -55,9 +55,20 @@ The core mechanics of Datastar's SSE support is
...
-### `ServerSentEventGenerator.send(eventType: EventType, dataLines: string[], options?: {id?:string, rery?:durationInMilliseconds})`
+### `ServerSentEventGenerator.send`
 
-All top level `ServerSentEventGenerator` ***should*** use a unified sending function.  This method ***should be private***
+```
+ServerSentEventGenerator.send(
+    eventType: EventType, 
+    dataLines: string[], 
+    options?: {
+        id?: string,
```
Comment (Superpat) [inline design/SDK.md:65]: I think this should be eventId to keep with the term used in the other methods.

File: design/SDK.md [hunk 2]
```diff
@@ -85,7 +96,21 @@ When called the function ***must*** write to the response buffer the following i
...
 5.  Afterward the writer ***should*** immediately flush.  This can be confounded by other middlewares such as compression layers
 
-### `ServerSentEventGenerator.RenderFragment(data: string, options ?: { selector?: string, merge ?: FragmentMergeMode, settleDuration?: durationInMilliseconds, useViewTransition?: boolean })`
+### `ServerSentEventGenerator.RenderFragment`
+
+```
+ServerSentEventGenerator.RenderFragment(
+    data: string, 
+    options?: { 
+        selector?: string,
```
Comment (Superpat) [inline design/SDK.md:105]: I'm being pedantic, but I wonder if we should change this to cssSelector

File: design/SDK.md [hunk 3]
```diff
@@ -85,7 +96,21 @@ When called the function ***must*** write to the response buffer the following i
...
 
-### `ServerSentEventGenerator.RenderFragment(data: string, options ?: { selector?: string, merge ?: FragmentMergeMode, settleDuration?: durationInMilliseconds, useViewTransition?: boolean })`
+### `ServerSentEventGenerator.RenderFragment`
+
+```
+ServerSentEventGenerator.RenderFragment(
+    data: string, 
+    options?: { 
+        selector?: string, 
+        merge?: FragmentMergeMode,
```
Comment (Superpat) [inline design/SDK.md:106]: And this makes it sound like a boolean, maybe mergeMode would be better ?

File: design/SDK.md [hunk 4]
```diff
@@ -85,7 +96,21 @@ When called the function ***must*** write to the response buffer the following i
...
+```
+ServerSentEventGenerator.RenderFragment(
+    data: string, 
+    options?: { 
+        selector?: string, 
+        merge?: FragmentMergeMode, 
+        settleDuration?: durationInMilliseconds, 
+        useViewTransition?: boolean,
+        eventId?: string, 
+        retry?: durationInMilliseconds
```
Comment (Superpat) [inline design/SDK.md:110]: In keeping with settleDuration. Maybe this should be retryEvery or retryDuration.

File: design/SDK.md [hunk 5]
```diff
@@ -120,8 +145,19 @@ When called the function ***must*** call `ServerSentEventGenerator.send` with th
 3. If the `settleDuration` is provided the function ***must*** include the settle duration in the event data in the format `settleDuration: DURATION`, unless the settle duration is the default of `300 milliseconds`.
 4. If the `useViewTransition` is provided the function ***must*** include the view transition in the event data in the format `useViewTransition VIEW_TRANSITION`, unless the view transition is the default of `false`.  `VIEW_TRANSITION` should be `true` or `false` depending on the value of the `useViewTransition` option.
 
-### `ServerSentEventGenerator.RemoveFragments(seletor: string, options ?: { settleDuration?: durationInMilliseconds, useViewTransition?: boolean})`
-
+### `ServerSentEventGenerator.RemoveFragments`
+
+```
+ServerSentEventGenerator.RemoveFragments(
+    seletor: string,
```
Comment (Superpat) [inline design/SDK.md:152]: This should be selector

**Review by Superpat — CHANGES_REQUESTED** [review]
Some of these are just suggestions

File: design/SDK.md (hunk 1 above)
Comment (bencroker) [reply design/SDK.md]: Done.

File: design/SDK.md (hunk 5 above)
Comment (bencroker) [reply design/SDK.md]: Fixed.

**Comment by bencroker** [comment]
@delaneyj Removed the references to an incrementing ID and added `retry` to the event logic.

Can you please respond to @Superpat ’s comments above?

File: design/SDK.md (hunk 4 above)
Comment (bencroker) [reply design/SDK.md:110]: I like `retryDuration`.

File: design/SDK.md (hunk 2 above)
Comment (delaneyj) [reply design/SDK.md:105]: it's not a CSS selector it's a DOM querySelector

File: design/SDK.md (hunk 3 above)
Comment (Superpat) [reply design/SDK.md:106]: Or just `mode`

Comment (Superpat) [reply design/SDK.md:106]: I'm thinking about dynamic languages mostly.

File: design/SDK.md (hunk 4 above)
Comment (bencroker) [reply design/SDK.md:110]: Renamed to `retryDuration`

File: design/SDK.md (hunk 2 above)
Comment (bencroker) [reply design/SDK.md:105]: I feel like `selector` is obvious enough in this case.

File: design/SDK.md (hunk 3 above)
Comment (bencroker) [reply design/SDK.md]: Renamed to `mergeMode`.

**Comment by bencroker** [comment]
Ok, resolved everything, final review please.

**Review by delaneyj — APPROVED** [review]

**Pull request merged**